### ISSUE TITLE
[a11y]: Nested Unordered ListItem bullet is read aloud

### ISSUE THREAD

**Issue by st004547** [carbon-design-system/carbon#10782]
### Package

carbon-components-react

### Browser

Chrome

### Operating System

Windows

### Package version

v7.52.0 (v10.52.0)

### React version

v17.0.2

### Automated testing tool and ruleset

Manual Test (AVT3)

### Assistive technology

JAWS, Windows Narrator

### Description

The screen reader will read aloud the content of the `::before` pseudo-element. In the case of an unordered list item, it will say "black small square" for each list item. 

The issue is described here: https://stackoverflow.com/questions/70224894/how-to-set-aria-hidden-to-pseudo-element-before?noredirect=1&lq=1

The simplest solution is to change the content of the pseudo-element to `content: "▪" / "";` as suggested by the [CSS Level 3 spec](https://drafts.csswg.org/css-content/#valdef-content---string--counter), but this feature is not yet supported on [Firefox](https://bugzilla.mozilla.org/show_bug.cgi?id=1281158).

### WCAG 2.1 Violation

1.3.1

### CodeSandbox example

https://75hin9.csb.app/

### Steps to reproduce

Open the code sandbox sample. Enable your screen reader. When it reads the heading, press the down arrow to move to the next item. It will say "black small square List item 1". The expectation is that is will just say "List item 1"

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
[Caniuse](https://caniuse.com/mdn-css_properties_content_alt_text) shows the `content` fix won't work in Safari either, though this problem may not extend beyond Windows/JAWS. We should test to see if voiceover on Mac/iOS reads these as well.

Another option from reading around on stackoverflow would be to provide a datauri svg for the bullet instead of the unicode character.

**Comment by dakahn** [comment]
Happy to go with the alt text solution. Like @st004547 mentioned this wont cover Firefox, but will provide a better experience for the other browsers until Mozilla catch up  👍🏾

**Comment by dakahn** [comment]
The alt text fix can't be used because Firefox not only doesn't support alt text in content, but sees its use as invalid CSS and ignores the whole attribute completely.

That said, I don't agree this constitutes a 1.3.1 violation. Nested list items being differentiated and described as being preceded by a "black small square" could be potentially critical for a user to understand that list item's position in relation to the parent list. This is from [_Understanding Success Criterion 1.3.1: Info and Relationships_](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)
> Sighted users perceive structure and relationships through various visual cues...headings are often in a larger, bold font separated from paragraphs by blank lines; list items are preceded by a bullet and perhaps indented...Having these structures and these relationships programmatically determined or available in text ensures that information important for comprehension will be perceivable to all

It's true our system's use of "black small square" is longer than some alternatives--"bullet" for example, but it _is still our "visual cue"_. Worth noting is that these bullets, squares and other generated content can be excluded or included purposefully based on the user's screen reader verbosity settings and browser pairing. 

It's worth noting that the default verbosity settings for VoiceOver in safari read "bullet" before every list item on the [WAI tutorial site for Content Structure Accessibility](https://www.w3.org/WAI/tutorials/page-structure/content/)
![image](https://user-images.githubusercontent.com/40970507/167073404-64756b3e-1cbd-4756-b760-07b0201bda15.png)

I'm closing this as resolved, but I'm happy to open it up again if there's something I missed.
